Automattic/wp-calypso
# Separate sites selectors.

Sites selectors all lived in a massive 1100+ line file, which made it very hard to tree-shake effectively.

This PR makes them all live in their own module, which can be imported separately. To ease migration, there's still an index file which imports and reexports them all.

This shouldn't have any immediate performance benefits, and the PR may actually slightly increase bundle size, due to the extra modules. It is expected to serve as the basis for future perf improvements, however.

Any perf improvements will likely only take place after dependants start importing individual selector modules, rather than importing from the index.

#### Changes proposed in this Pull Request

* Separate all the selectors in `client/state/sites` into their own modules.

#### Testing instructions

* Use Calypso normally and look out for any issues. There are no code changes to the selectors, so there shouldn't be any behaviour-type bugs introduced. Any bugs would likely be caused by an incorrect import somewhere. E2E tests should help here.
